src/Symfony/Bridge/Doctrine/Form/DoctrineOrmTypeGuesser.php

```diff
@@ -56,31 +57,32 @@ public function guessType($class, $property)
             return new TypeGuess('Symfony\Bridge\Doctrine\Form\Type\EntityType', ['em' => $name, 'class' => $mapping['targetEntity'], 'multiple' => $multiple], Guess::HIGH_CONFIDENCE);
         }
 
+        $useDeprecatedConstants = !class_exists(Types::class);
         switch ($metadata->getTypeOfField($property)) {
-            case Type::TARRAY:
+            case $useDeprecatedConstants ? Type::TARRAY : 'array':
```

Comment:
A constant named `ARRAY` cannot work on PHP 5 so we have to hardcode it here I guess? And then use `Types::ARRAY` on 4.3+.